langchain/callbacks/arize_callback.py

```diff
@@ -1,217 +1,314 @@
-                    vector_column_name="prompt_vector", data_column_name="prompt"
-                )
-
-                response_columns = EmbeddingColumnNames(
-                    vector_column_name="response_vector", data_column_name="response"
-                )
-
-                schema = Schema(
-                    timestamp_column_name="prediction_ts",
-                    tag_column_names=[
-                        "prompt_token",
-                        "completion_token",
-                        "total_token",
-                    ],
-                    prompt_column_names=prompt_columns,
-                    response_column_names=response_columns,
-                )
-
-                response_from_arize = self.arize_client.log(
-                    dataframe=df,
-                    schema=schema,
-                    model_id=self.model_id,
-                    model_version=self.model_version,
-                    model_type=ModelTypes.GENERATIVE_LLM,
-                    environment=Environments.PRODUCTION,
-                )
-                if response_from_arize.status_code == 200:
-                    print("✅ Successfully logged data to Arize!")
-                else:
-                    print(f'❌ Logging failed "{response_from_arize.text}"')
-
-    def on_llm_error(
-        self, error: Union[Exception, KeyboardInterrupt], **kwargs: Any
-    ) -> None:
-        """Do nothing."""
-        pass
+
+def _import_package(package_name: str) -> ModuleType:
+    try:
+        package = importlib.import_module(package_name)
+    except ImportError:
+        raise ImportError(f"The {package_name} package must be installed.")
+    return package
+
+
+def validate_package_version(
+    package: ModuleType,
+    min_supported_version: Optional[str],
+    max_supported_version: Optional[str],
+) -> None:
+    # TODO: implement package version validation at runtime
+    ...
+
+
+class BaseArizeCallbackHandler(BaseCallbackHandler, ABC):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._chain_data: Optional[ChainData] = None
+        self._base_chain_name: Optional[str] = None
 
     def on_chain_start(
-        self, serialized: Dict[str, Any], inputs: Dict[str, Any], **kwargs: Any
+        self,
+        serialized: Dict[str, Any],
+        inputs: Dict[str, Any],
+        *,
+        run_id: UUID,
+        parent_run_id: Optional[UUID] = None,
+        tags: Optional[List[str]] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
     ) -> None:
-        pass
-
-    def on_chain_end(self, outputs: Dict[str, Any], **kwargs: Any) -> None:
-        """Do nothing."""
-        pass
+        chain_name = ".".join(serialized["id"])
+        if self._is_base_chain(parent_run_id):
+            if not is_retrieval_qa_chain_type(chain_name):
+                raise ValueError(f"Unsupported chain type: {chain_name}.")
```

Comment:
I think there might be a consideration to not raise errors and rather just log an error. I'm not sure what LangChain's practice is on this but I think this might lead to developer frustration.